File: src/main/java/com/appointmed/user/RegisterRequest.java
package com.appointmed.user;

public class RegisterRequest {

    private String fullName;
    private String email;
    private String password;
    private String role;

    // Additional info
    private String address;
    private String gender;
    private String birthDate;

    // Doctor-only fields
    private String specialization;
    private String licenseNumber;
    private String phone;
    private String clinicAddress;

    public RegisterRequest() {
    }

    public String getFullName() {
        return fullName;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public String getRole() {
        return role;
    }

    public String getSpecialization() {
        return specialization;
    }

    public String getLicenseNumber() {
        return licenseNumber;
    }

    public String getAddress() {
        return address;
    }

    public String getGender() {
        return gender;
    }

    public String getBirthDate() {
        return birthDate;
    }

    public String getPhone() {
        return phone;
    }

    public String getClinicAddress() {
        return clinicAddress;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public void setBirthDate(String birthDate) {
        this.birthDate = birthDate;
    }

    public void setSpecialization(String specialization) {
        this.specialization = specialization;
    }

    public void setLicenseNumber(String licenseNumber) {
        this.licenseNumber = licenseNumber;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setClinicAddress(String clinicAddress) {
        this.clinicAddress = clinicAddress;
    }
}

File: mobile/app/src/main/java/com/appointmed/mobile/data/model/RegisterRequest.kt
package com.appointmed.mobile.data.model

import com.google.gson.annotations.SerializedName

data class RegisterRequest(
    @SerializedName("fullName") val fullName: String,
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("role") val role: String,
    @SerializedName("address") val address: String? = null,
    @SerializedName("gender") val gender: String? = null,
    @SerializedName("birthDate") val birthDate: String? = null,
    @SerializedName("specialization") val specialization: String? = null,
    @SerializedName("licenseNumber") val licenseNumber: String? = null,
    @SerializedName("phone") val phone: String? = null,
    @SerializedName("clinicAddress") val clinicAddress: String? = null,
    @SerializedName("consultationFee") val consultationFee: Int? = null
)
